File: python/ui/NumbersPane.py
# coding=utf-8

import Tkinter as tk
from WorkPanel import WorkPanel

class NumbersPane(tk.Frame):
    def __init__(self,master):
        tk.Frame.__init__(self,master,name="numbers-pane")

        self.grid(row=0, sticky=(tk.N+tk.E+tk.W+tk.S))
        for col in xrange(2):
            self.grid_columnconfigure(col, weight=1)
        for row in xrange(4):
            self.grid_rowconfigure(row, weight=1)

        self.lblInit = tk.Label(self,text="Inicio:")
        self.lblInit.grid(row=0,column=0,sticky=tk.W)

        self.txtInit = tk.StringVar(self, "")
        self.entryInit = tk.Entry(self,textvariable=self.txtInit,width=3)
        self.entryInit.grid(row=0,column=1,sticky=tk.E+tk.W)

        self.lblEnd = tk.Label(self,text="Fin:")
        self.lblEnd.grid(row=1,column=0,sticky=tk.W)

        self.txtEnd = tk.StringVar(self, "")
        self.entryEnd = tk.Entry(self,textvariable=self.txtEnd,width=3)
        self.entryEnd.grid(row=1, column=1, sticky=tk.E+tk.W)

        self.lblNumbersXSheet = tk.Label(self,text="Números X Hoja:")
        self.lblNumbersXSheet.grid(row=2,column=0,sticky=tk.W)

        self.txtNumbersXSheet = tk.StringVar(self,"")
        self.entryNumbersXSheet = tk.Entry(
            self, textvariable=self.txtNumbersXSheet, width=3)
        self.entryNumbersXSheet.grid(row=2, column=1, sticky=tk.E+tk.W)

        self.lblUseZero = tk.Label(self, text="Agregar ceros:")
        self.lblUseZero.grid(row=3, column=0,sticky=tk.W)

        self.blnUseZero = tk.BooleanVar(self,False)
        self.chkUseZero = tk.Checkbutton(self,variable=self.blnUseZero,onvalue=True,offvalue=False)
        self.chkUseZero.grid(row=3,column=1,sticky=tk.W)

    def getText(self):
        if self.txtInit.get() == "":
            self.txtInit.set("0")
        if self.txtEnd.get() == "":
            self.txtEnd.set("0")
        if self.txtNumbersXSheet.get() == "":
            self.txtNumbersXSheet.set("1")
        
        init = int(self.txtInit.get())
        end  = int(self.txtEnd.get())
        nXs  = int(self.txtNumbersXSheet.get())
        if nXs <= 0:
            nXs = 1
        total = end - init + 1
        numSheets = total / nXs

        if total%nXs!=0:
            numSheets += 1
        
        header_list = ["c{0}".format(cell) for cell in xrange(1,nXs+1)]
        header = ",".join(header_list)

        leading_zero_format = ""
        if self.blnUseZero.get():
            leading_zero_format = ":0{LEN}".format(LEN=len(self.txtEnd.get()))
        
        number_format = "{{0{LEADING_FORMAT}}}".format(LEADING_FORMAT=leading_zero_format)
        
        numbers_list = [",".join([number_format.format((init+row)+(numSheets*cell)) for cell in xrange(nXs)]) for row in xrange(numSheets)]
        numbers_string = "\n".join(numbers_list)

        return header+"\n"+numbers_string

File: java/ui/NumbersPane.java
package ui;

import java.util.Formatter;
import java.awt.GridLayout;
import javax.swing.JPanel;
import javax.swing.JTextField;
import javax.swing.JLabel;
import javax.swing.JCheckBox;

public class NumbersPane extends JPanel implements WorkPanel{
	private JTextField	txtInit;
	private JTextField	txtEnd;
	private JTextField	txtNumbersXSheet;
	private JCheckBox	chkUseZero;
	private JLabel		lblInit;
	private JLabel		lblEnd;
	private JLabel		lblNumbersXSheet;
	private JLabel		lblUseZero;

	public NumbersPane(){
		super();

		setLayout(new GridLayout(4,2));
		lblInit = new JLabel("Inicio:");
		lblEnd = new JLabel("Fin:");
		lblNumbersXSheet = new JLabel("Numeros X Hoja:");
		lblUseZero = new JLabel("Agregar ceros:");
		txtInit = new JTextField(3);
		txtEnd = new JTextField(3);
		txtNumbersXSheet = new JTextField(3);
		chkUseZero = new JCheckBox();

		add(lblInit);
		add(txtInit);
		add(lblEnd);
		add(txtEnd);
		add(lblNumbersXSheet);
		add(txtNumbersXSheet);
		add(lblUseZero);
		add(chkUseZero);
	}

	@Override
	public String getText(){
		int init = Integer.parseInt(txtInit.getText());
		int end  = Integer.parseInt(txtEnd.getText());
		int nXs  = Integer.parseInt(txtNumbersXSheet.getText());
		int total = end - init + 1;
		int numSheets = total / nXs;
		if(total%nXs!=0)
			numSheets++;

		StringBuilder sb = new StringBuilder();
		Formatter formatter = new Formatter(sb);
		for(int cell=1;cell<=nXs;cell++){
			formatter.format("c%d",cell);
			if(cell<nXs)
				sb.append(",");
		}

		sb.append("\n");
		String numberFormat = "%"+(chkUseZero.isSelected()?"0":"")+txtEnd.getText().length()+"d";

		for(int row=0;row<numSheets;row++){
			for(int cell=0;cell<nXs;cell++){
				formatter.format(numberFormat,(init+row)+(numSheets*cell));
				if(cell<nXs-1)
					sb.append(",");
			}
			sb.append("\n");
		}

		return sb.toString();
	}
}